### DISCUSSION TITLE
Update postgresql array column

### DISCUSSION THREAD

**Discussion by MijikHna** [sqlalchemy/sqlalchemy#8447] [Usage Questions] [answered]
I have an orm class 

```
...
from sqlalchemy.dialects import postgresql
from sqlalchemy import ..., ARRAY, String, cast

MyClass(Base):
  id: Column = Column(Integer, primary_key=True)
  my_array_column: Column = Column(
        postgresql.ARRAY(String),
        nullable=False,
        #server_default=postgresql.array([] type_=postgesql.CHAr),
        server_default=cast(array([]),  ARRAY(String)),
  )
```
in my service I want to append new value to this array. After flush it seems like this new value is in the array. But after commit I get empty array again. Also in db I have empty array.

```
...
my_class: MyClass = db_session.query(MyClass).filter(MyClass.id == id).one()
my_class.my_array_column.append('test')
db_session.flush()
...
db_session.comit()
```

if I print the type of `my_array_column` python says it's a `list`. So I thought sqlalchemy do all the casts in background, but it seems not to be the case. Also I haven't found in the documentation how I can add new values to the array.

**Comment by zzzeek** [comment, marked as answer]
if you want PG ARRAY with the ORM being notified of in-place mutations you need to use [MutableList](https://docs.sqlalchemy.org/en/14/orm/extensions/mutable.html?highlight=mutablelist#sqlalchemy.ext.mutable.MutableList)

incredibly, the docs for ARRAY don't mention this, which is a doc bug.  We have notes about Mutable all over the place so I can't believe we dont have on here.  

example

```py
from sqlalchemy.ext.mutable import MutableList

class MyClass(Base):
  id: Column = Column(Integer, primary_key=True)
  my_array_column: Column = Column(
        MutableList.as_mutable(postgresql.ARRAY(String)),
        nullable=False,
        #server_default=postgresql.array([] type_=postgesql.CHAr),
        server_default=cast(array([]),  ARRAY(String)),
  )
```

**Reply by MijikHna** [reply]
Thank you very much!. That was the solution

**Reply by zzzeek** [reply]
added documentation for this in 5972a06ea18434d620685eb
